### PR TITLE
zos: use PLO builtins for atomic operations

### PR DESCRIPTION
**Pull request by jBarz**
Use builtins provided that perform compare and swap operations
using the PLO instruction.

### CODE REVIEW

File: src/unix/atomic-ops.h [hunk 1]
```diff
@@ -68,13 +66,11 @@ UV_UNUSED(static long cmpxchgl(long* ptr, long oldval, long newval)) {
 # endif /* if defined(__64BIT__) */
   return out;
 #elif defined (__MVS__)
-  const long out = (*(volatile int*) ptr);
 # ifdef _LP64
-  cds((cds_t*)(&oldval), (cds_t*)ptr, *(cds_t*)(&newval));
+  return __plo_CSGR(ptr, (unsigned long long*) ptr, oldval, (unsigned long long*) &newval);
```
Comment (iWuzHere) [inline src/unix/atomic-ops.h]: line is greater than 80 chars, please wrap.

**Comment by iWuzHere** [comment]
Commit subject is greater than 50 chars.

LGTM with nits

**Comment by iWuzHere** [comment]
Changing the title on github does not change your commit message.
You will have to interactive rebase, use fixup and amend then force push

**Comment by Fishrock123** [comment]
(`git commit --amend`, `git push origin head --force`)

**Comment by jBarz** [comment]
oops! This is what happens when I work late at night.  :)

**Comment by saghul** [comment]
LGTM (I take your word for it :-) )

File: src/unix/atomic-ops.h [hunk 2]
```diff
@@ -68,13 +67,13 @@ UV_UNUSED(static long cmpxchgl(long* ptr, long oldval, long newval)) {
 # endif /* if defined(__64BIT__) */
   return out;
 #elif defined (__MVS__)
-  const long out = (*(volatile int*) ptr);
 # ifdef _LP64
-  cds((cds_t*)(&oldval), (cds_t*)ptr, *(cds_t*)(&newval));
+  return __plo_CSGR(ptr, (unsigned long long*) ptr,
+                    oldval, (unsigned long long*) &newval);
```
Comment (bnoordhuis) [inline src/unix/atomic-ops.h]: Is `sizeof(long) == sizeof(unsigned long long)` on 64 bits z/os?

**Comment by bnoordhuis** [comment]
LGTM with a question.

File: src/unix/atomic-ops.h (hunk 2 above)
Comment (jBarz) [inline src/unix/atomic-ops.h]: that is correct

**Comment by saghul** [comment]
Landed in 841edfc, thanks! 🎉

**Pull request closed without merging**